packages/manager/apps/dedicated/client/app/account/user/infos/user-infos.controller.js

```diff
@@ -212,6 +214,22 @@ angular.module('UserAccount').controller('UserAccount.controllers.Infos', [
       loadTaskForEmailValidation($scope.controls.validateEmailChange.taskId);
     };
 
+    $scope.requestChangeEmailToken = function requestChangeEmailToken(email) {
+      $scope.requestingToken = true;
+      return UseraccountInfos.changeEmail(email)
+        .then(({ id }) => {
+          $scope.controls.taskEmailChangeTodo.id = id;
+          // reset requestingToken after a long delay to avoid the user
+          // being tempted to make too many token requests
+          $timeout(() => { $scope.requestingToken = false; }, 60000);
```

Comment:
This should be handled by the API, not by us

Reply:
Not really, it's just a frontside trick to avoid the user being tempted to click repeatedly on the button and make him wait (because the email might takes some time to be received).
It's not a security/limitation feature implemented on api side.

Reply:
Maybe we could see to have a task to poll instead or something like that ?

Reply:
it's not possible, we cannot know when the user will receive the email (network latency etc)

Reply:
Why does it matter if the user clicks repeatedly ? 🤔 I feel like it shouldn't matter to us

If it really matters because the API can't handle it, just disable the button, what do you think ?  :)